provision/run-provisioners.sh

```diff
@@ -0,0 +1,242 @@
+#!/bin/bash
+
+ORIGINAL_STDOUT=$(readlink -f /proc/$$/fd/1)
+ORIGINAL_STDERR=$(readlink -f /proc/$$/fd/2)
+
+source /srv/provision/provision-helpers.sh
+
+get_config_value() {
+  local value=$(shyaml get-value "${1}" 2> /dev/null < ${VVV_CONFIG})
+  echo "${value:-$2}"
+}
+
+get_config_values() {
+  local value=$(shyaml get-values "${1}" 2> /dev/null < ${VVV_CONFIG})
+  echo "${value:-$2}"
+}
+
+get_config_type() {
+  local value=$(shyaml get-type "${1}" 2> /dev/null < ${VVV_CONFIG})
+  echo "${value}"
+}
+
+get_config_keys() {
+  local value=$(shyaml keys "${1}" 2> /dev/null < ${VVV_CONFIG})
+  echo "${value:-$2}"
+}
+
+provisioner_begin() {
+  log_to_file "provisioner-${1:-${FUNCNAME[1]}}"
+  touch "/vagrant/failed_provisioners/provisioner-${1:-${FUNCNAME[1]}}"
+  PROVISION_SUCCESS="1"
+  echo -e "------------------------------------------------------------------------------------"
+  echo -e "${GREEN} ▷ Running the '${1:-${FUNCNAME[1]}}' provisioner...${CRESET}"
+  echo -e "------------------------------------------------------------------------------------"
+  start_seconds="$(date +%s)"
+}
+
+provisioner_end() {
+  end_seconds="$(date +%s)" 
+  local elapsed="$(( end_seconds - start_seconds ))"
+  if [[ $PROVISION_SUCCESS -eq "0" ]]; then
+    echo -e "------------------------------------------------------------------------------------"
+    echo -e "${GREEN} ✔ The '${1:-${FUNCNAME[1]}}' provisioner completed in ${elapsed} seconds.${CRESET}"
+    echo -e "------------------------------------------------------------------------------------"
+    rm -f "/vagrant/failed_provisioners/provisioner-${1:-${FUNCNAME[1]}}"
+  else
+    echo -e "------------------------------------------------------------------------------------"
+    echo -e "${RED} ! The '${1:-${FUNCNAME[1]}}' provisioner ran into problems, check the full log for more details! It completed in ${elapsed} seconds.${CRESET}"
+    echo -e "------------------------------------------------------------------------------------"
+  fi
+  echo ""
+}
+
+provisioner_init() {
+  # fix no tty warnings in provisioner logs
+  sudo sed -i '/tty/!s/mesg n/tty -s \\&\\& mesg n/' /root/.profile
+
+  export DEBIAN_FRONTEND=noninteractive
+  export APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1
+  export COMPOSER_ALLOW_SUPERUSER=1
+  export COMPOSER_NO_INTERACTION=1
+
+  # cleanup
+  mkdir -p /vagrant
+  mkdir -p /vagrant/failed_provisioners
+
+  rm -f /vagrant/provisioned_at
+  rm -f /vagrant/version
+  rm -f /vagrant/vvv-custom.yml
+  rm -f "${VVV_CONFIG}"
+
+  touch /vagrant/provisioned_at
+  echo $(date "+%Y.%m.%d_%H-%M-%S") > /vagrant/provisioned_at
+
+  # copy over version and config files
+  cp -f /home/vagrant/version /vagrant
+  cp -f /srv/config/config.yml "${VVV_CONFIG}"
+
+  sudo chmod 0644 "${VVV_CONFIG}"
+  sudo chmod 0644 /vagrant/version
+  sudo chmod 0644 /vagrant/provisioned_at
+
+  # change ownership for /vagrant folder
+  sudo chown -R vagrant:vagrant /vagrant
```

Comment:
We may need to create the `vagrant` user, or, create our own user and add it to the necessary groups. If we run this in a docker container or on a host, `vagrant` won't be available as a user/group

Reply:
Got that covered in another branch I have locally, where the user to use is settable on the main provisioner.